server/server.go

```diff
@@ -53,38 +59,26 @@ func (cs *cribbageServer) Serve() {
 }
 
 func (cs *cribbageServer) ginPostCreateGame(c *gin.Context) {
-	var pIDs []model.PlayerID
-
-	pID := getPlayerID(c, `player1`)
-	if pID == model.InvalidPlayerID {
-		c.String(http.StatusBadRequest, `Needs player1`)
-		return
-	}
-	pIDs = append(pIDs, pID)
-
-	pID = getPlayerID(c, `player2`)
-	if pID == model.InvalidPlayerID {
-		c.String(http.StatusBadRequest, `Needs player2`)
+	var gameReq model.Game
+	err := c.ShouldBindJSON(&gameReq)
```

Comment:
👓 🚒 I'm not convinced this is going to work. Our `PlayerAction` has an interface, so we need to use this everywhere else: 

https://github.com/joshprzybyszewski/cribbage/blob/252193173385e65ca77379e3a621daf9bf8aa0b8/jsonutils/gameUnmarshaller.go#L9-L12

This `ShouldBindJSON` may work for "create game" because there should be no actions, but that assumes we'll never want an action on game creation. it's risky.

What other options do we have?

Reply:
I've created a custom implementation of gin's `binding.Binding` and `binding.BindingBody` interfaces. I still need to write some unit tests, but it should do what we want.

https://github.com/joshprzybyszewski/cribbage/blob/7d834ebb8660968768c43fc1b0aeff03d1dec050/jsonutils/gameBinding.go#L1-L46

Reply:
And now I have some tests 🧯